src/forms/FormFieldUpload/FormFieldUpload.tsx

```diff
@@ -200,8 +200,14 @@ const FormFieldUpload = (props: MosaicFieldProps<"upload", UploadFieldInputSetti
 	};
 
 	const shouldDisableField = useMemo(() => {
-		return fieldDef.disabled || (limit !== undefined && value?.length >= limit)
-	}, [fieldDef.disabled, limit, value]);
+		const numOfPendingFiles = Object.keys(pendingFiles).length;
+
+		return (
+			fieldDef.disabled ||
+			(limit !== undefined && value?.length >= limit ||
```

Comment:
I think there's some repeated code here that could be collapsed. If we think of these "or" statements as venn diagrams we have 3 intersects, which could be combined into 1 single operation.
```
return (
			fieldDef.disabled ||
			(
				limit !== undefined
				&& limit >= 0
				&& (value !== undefined ? value.length : 0) + numOfPendingFiles >= limit
			)
		)
```

pendingFiles will always be an object so it's length will always be a number, the only thing we'd need to check for would be if value exists, and if it does, we put it's length, if not then it's a 0